Tribler/community/tunnel/tunnel_community.py

```diff
@@ -609,12 +630,23 @@ def remove_relay(self, circuit_id, additional_info='', destroy=False, got_destro
                 self.tunnel_logger.error("Could not remove relay %d %s", circuit_id, additional_info)
 
     def remove_exit_socket(self, circuit_id, additional_info='', destroy=False):
+
+        self.tunnel_logger.error("MULTICHAIN: Removing exit socket")
+
         if circuit_id in self.exit_sockets:
             if destroy:
                 self.destroy_exit_socket(circuit_id)
 
             # Close socket
             exit_socket = self.exit_sockets.pop(circuit_id)
+            if self.notifier:
+                peer = (exit_socket.sock_addr[0], exit_socket.sock_addr[1])
+                candidate = self.get_candidate(peer)
+                if candidate:
+                    from Tribler.Core.simpledefs import NTFY_TUNNEL, NTFY_REMOVE
```

Comment:
This is done to allow the community to run without the Tribler code being present.